packages/manager/src/managers/screenshots/ScreenshotsManager.ts

```diff
@@ -118,9 +118,10 @@ export class ScreenshotsManager extends BaseManager {
 	}
 
 	async initS3ACL(): Promise<void> {
-		if (this._s3ACL) {
-			return;
-		}
+		// TODO: we need to find a way to create a new AWS ACL only when necessary (e.g., when it has expired).
+		// if (this._s3ACL) {
+		// 	return;
+		// }
```

Comment:
I assume this would require an endpoint on our S3 API to validate the session, maybe we could add a small cache/debounce here (like a 60s cache) in the meantime to prevent spamming the S3 ACL init endpoint when we're uploading all screenshots at once.

We're doing something similar on the client to dedupe `/api/v2` request (a bit hard to follow at first I admit): https://github.com/prismicio/prismic-client/blob/v7/src/createClient.ts#L1858-L1910